### PR TITLE
TS-4652 Fixes log field initialization, previously crippled

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

**Comment by zwoop** [comment]
@bryancall  Please review.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/452/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/347/ for details.

File: proxy/logging/Log.cc [hunk 1]
```diff
@@ -925,6 +925,7 @@ Log::init(int flags)
     if (config_flags & STANDALONE_COLLATOR) {
       config->collation_mode = Log::COLLATION_HOST;
     }
+    init_fields();
```
Comment (jpeach) [inline proxy/logging/Log.cc]: Both sided of this branch now call ``init_fields()``. Do you want to hoist the call up and simplify?

**Comment by zwoop** [comment]
Let me take a look, but I was going for the simplest solution that works, with minimal risk. It's pretty finicky on the order  :)

**Comment by zwoop** [comment]
Ah, yeah, I'm pretty sure that entire if() can be safely eliminated (famous last words ...).

**Comment by zwoop** [comment]
@jpeach @bryancall check the update please.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/456/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/351/ for details.

**Comment by jpeach** [comment]
👍

**Comment by bryancall** [comment]
👍

**Pull request merged**